lib/Modules/HyperVCIHooks/HyperVCIHooks.psm1

```diff
@@ -0,0 +1,1482 @@
+#
+# Copyright 2015-2016 Cloudbase Solutions Srl
+#
+
+$ErrorActionPreference = 'Stop'
+
+Import-Module JujuUtils
+Import-Module JujuHooks
+Import-Module JujuLogging
+Import-Module JujuHelper
+Import-Module JujuWindowsUtils
+Import-Module ADCharmUtils
+Import-Module Templating
+
+$NEUTRON_GIT           = "https://github.com/openstack/neutron.git"
+$NOVA_GIT              = "https://github.com/openstack/nova.git"
+$NETWORKING_HYPERV_GIT = "https://github.com/openstack/networking-hyperv.git"
+$COMPUTE_HYPERV_GIT    = "https://github.com/openstack/compute-hyperv.git"
+$OSWIN_GIT             = "https://git.openstack.org/openstack/os-win.git"
+
+$OPENSTACK_DIR  = Join-Path $env:SystemDrive "OpenStack"
+$PYTHON_DIR     = Join-Path $env:SystemDrive "Python27"
+$LIB_DIR        = Join-Path $PYTHON_DIR "lib\site-packages"
+$BUILD_DIR      = Join-Path $OPENSTACK_DIR "build"
+$INSTANCES_DIR  = Join-Path $OPENSTACK_DIR "Instances"
+$BIN_DIR        = Join-Path $OPENSTACK_DIR "bin"
+$CONFIG_DIR     = Join-Path $OPENSTACK_DIR "etc"
+$LOG_DIR        = Join-Path $OPENSTACK_DIR "log"
+$SERVICE_DIR    = Join-Path $OPENSTACK_DIR "service"
+$FILES_DIR      = Join-Path ${env:CHARM_DIR} "files"
+$OVS_DIR        = "${env:ProgramFiles}\Cloudbase Solutions\Open vSwitch"
+$OVS_VSCTL      = Join-Path $OVS_DIR "bin\ovs-vsctl.exe"
+$env:OVS_RUNDIR = Join-Path $env:ProgramData "openvswitch"
+$OVS_EXT_NAME   = "Open vSwitch Extension"
+$INTERFACES_TEMPLATE = Join-Path $CONFIG_DIR "interfaces.template"
+$POLICY_FILE = Join-Path $CONFIG_DIR "policy.json"
+$MKISO_EXE = Join-Path $BIN_DIR "mkisofs.exe"
+$QEMU_IMG_EXE = Join-Path $BIN_DIR "qemu-img.exe"
+
+function Get-TemplatesDir {
+    return (Join-Path (Get-JujuCharmDir) "templates")
+}
+
+
+function Get-DevStackContext {
+    $requiredCtx =  @{
+        "devstack_ip" = $null;
+        "password"    = $null;
+        "rabbit_user" = $null;
+    }
+    $ctx = Get-JujuRelationContext -Relation 'devstack' -RequiredContext $requiredCtx
+
+    # Required context not found
+    if(!$ctx.Count) {
+        return @{}
+    }
+
+    return $ctx
+}
+
+
+function Get-SystemContext {
+    $systemCtxt = @{
+        "instances_path"      = $INSTANCES_DIR;
+        "interfaces_template" = $INTERFACES_TEMPLATE;
+        "policy_file"         = $POLICY_FILE;
+        "mkisofs_exe"         = $MKISO_EXE;
+        "log_directory"       = $LOG_DIR;
+        "qemu_img_exe"        = $QEMU_IMG_EXE;
+        "compute_driver"      = Get-ComputeDriver
+        "vswitch_name"        = Get-JujuVMSwitchName
+        "local_ip"            = (Get-CharmState -Namespace "novahyperv" -Key "local_ip");
```

Comment:
Is this namespace not identical with the "standard" nova hyper-v charm? I'd replace all it's references with a namespace name related to this charm (compute charm from source code on hyper-v, it's not "nova on hyper-v").